### PR TITLE
Added graceful exit if gcore process exits with error

### PR DESCRIPTION
**Pull request by adamkimsft**
Wait for gcore child process to exit, check exit status, and if non-zero, output messages from stdout and stderr of child process and exit with error code.

### CODE REVIEW

File: src/CoreDumpWriter.c [hunk 1]
```diff
@@ -506,13 +506,18 @@ int WriteCoreDumpInternal(struct CoreDumpWriter *self, char* socketName)
         // someone wants to use gcore even for .NET Core 3.x+ processes.
         commandPipe = popen2(command, "r", &gcorePid);
         self->Config->gcorePid = gcorePid;
-        
+
+        // Wait for child process to end and get exit status for bash gcore command
+        int stat;
+        waitpid(gcorePid, &stat, 0);
+        int gcoreStatus = WEXITSTATUS(stat);
```
Comment (josalem) [inline src/CoreDumpWriter.c:513]: This should come after we read the stream.

Normally, you would use [`pclose`](https://www.man7.org/linux/man-pages/man3/popen.3.html) for this, which should also block till the process is done and give the return code.

N.B. - the exit code will be the exit code of the process, but can also be `-1` if `pclose` itself fails.

File: src/CoreDumpWriter.c [hunk 2]
```diff
@@ -544,7 +549,7 @@ int WriteCoreDumpInternal(struct CoreDumpWriter *self, char* socketName)
                 }
             }
 
-            exit(1);
+            exit(gcoreStatus > 0 ? gcoreStatus : 1);
```
Comment (josalem) [inline src/CoreDumpWriter.c:552]:
```suggestion
            exit(gcoreStatus != 0 ? gcoreStatus : 1);
```

File: src/CoreDumpWriter.c (hunk 1 above)
Comment (adamkimsft) [reply src/CoreDumpWriter.c]: I moved it in updated pull request.

File: src/CoreDumpWriter.c [hunk 3]
```diff
@@ -534,8 +539,8 @@ int WriteCoreDumpInternal(struct CoreDumpWriter *self, char* socketName)
         pclose(commandPipe);
```
[the remark this answers is not in the document]
Comment (adamkimsft) [reply src/CoreDumpWriter.c]: I also added a separate error check variable for pclose result.

File: src/CoreDumpWriter.c [hunk 4]
```diff
@@ -529,21 +529,38 @@ int WriteCoreDumpInternal(struct CoreDumpWriter *self, char* socketName)
...
         
+        // After reading all input, wait for child process to end and get exit status for bash gcore command
+        int stat;
+        waitpid(gcorePid, &stat, 0);
+        int gcoreStatus = WEXITSTATUS(stat);
+        
         // close pipe reading from gcore
         self->Config->gcorePid = NO_PID;                // reset gcore pid so that signal handler knows we aren't dumping
-        pclose(commandPipe);
+        int pcloseStatus = pclose(commandPipe);
```
Comment (josalem) [inline src/CoreDumpWriter.c:539]: I think these values are going to be the same, so we probably only need one or the other.

Comment (adamkimsft) [reply src/CoreDumpWriter.c:539]: So the return from closing the output pipe will be the same as the return from exit status from waitpid?

**Review by josalem — APPROVED** [review]

**Pull request merged**